automatic/nexus-repository/tools/chocolateyinstall.ps1

```diff
@@ -35,18 +42,89 @@ if (Test-Path "$env:ProgramFiles\nexus\bin") {
   Get-Service $ServiceName | Stop-Service -Force
 }
 
-if ($pp.ContainsKey("BackupSslConfig")) {
+if ($pp.ContainsKey("BackupSslConfig") -or $CurrentConfig.'application-port-ssl' -gt 0) {
   if ($pp.ContainsKey("BackupLocation")) {
     Backup-NexusSSL -BackupLocation $pp["BackupLocation"]
   } else {
     Backup-NexusSSL
   }
 }
 
+if ($CurrentlyInstalledVersion -and [version]($CurrentlyInstalledVersion -replace '-\d+$') -lt "3.70") {
+  Write-Error (@(
+    "Please upgrade nexus-repository to version 3.70.1-02 before upgrading further."
+    "You can do this by running 'choco upgrade nexus-repository --version 3.70.1.2 --confirm'"
+    "For more details, please see: https://help.sonatype.com/en/orient-pre-3-70-java-8-or-11.html"
+  ) -join "`n")
+  throw "Package cannot upgrade from '$($CurrentlyInstalledVersion)' to '$($env:ChocolateyPackageVersion)'"
+} elseif ($CurrentlyInstalledVersion -and [version]($CurrentlyInstalledVersion -replace '-\d+$') -eq "3.70.1" -and (Test-NexusDatabaseType -Type "OrientDb" -DataDir $TargetDataFolder -ProgramDir $TargetFolder)) {
+  Write-Host "Preparing for database migration..."
+  $MigrationFiles = Join-Path $ExtractFolder "dbmigration"
+  if ((Get-PSDrive $MigrationFiles[0]).Free -lt ($RequiredSpace = [Math]::Max((3 * (Get-ChildItem -Path $TargetDataFolder\nexus3\db -Recurse | Measure-Object Length -Sum).Sum), 10GB))) {
+    Write-Error -Message (@(
+      "The Sonatype Nexus Database Migrator requires at least $([Math]::Round(($RequiredSpace / 1GB), 2))GB free space."
+      "There is only $([Math]::Floor((Get-PSDrive $MigrationFiles[0]).Free / 1GB))GB free. For more details, please see: "
+      "https://help.sonatype.com/en/orient-3-70-java-8-or-11.html#considerations-before-migrating-252166"
+    ) -join "`n")
+    throw "Cannot migrate database with available disk space"
+  }
+  if (($Memory = Get-CimInstance Win32_PhysicalMemory).Capacity -lt 16GB) {
+    Write-Warning -Message (@(
+      "The Sonatype Nexus Database Migrator requires at least 16GB free space."
+      "There is only $($Memory.Capacity/1GB)GB available. For more details, please see: "
+      "https://help.sonatype.com/en/orient-3-70-java-8-or-11.html#considerations-before-migrating-252166"
+    ) -join "`n")
+  }
+  New-NexusOrientDbBackup -DataDir $TargetDataFolder -ServiceName $ServiceName -DestinationPath $MigrationFiles
+
+  # See: https://help.sonatype.com/en/orientdb-downloads.html
+  $MigratorDownload = @{
+    PackageName = $env:ChocolateyPackageName
+    Url = 'https://download.sonatype.com/nexus/nxrm3-migrator/nexus-db-migrator-3.70.1-03.jar'
+    FileFullPath = Join-Path $toolsDir "nexus-db-migrator.jar"
+    Checksum = "02ea88e88e7d5d7f9d8a1738c9c0363fde636fc22e295efddf9d0f5e7bed982a"
+    ChecksumType = "SHA256"
+  }
+  Get-ChocolateyWebFile @MigratorDownload
+
+  try {
+    Push-Location $MigrationFiles
+    Write-Host "Migrating database from 'OrientDb' to 'H2'"
+    $JavaPath = Join-Path $TargetFolder "jre/bin/java.exe"
```

Comment:
Is this java binary always going to be java 8/11, i.e. a compatible version with the migrator `.jar`?

Reply:
Yes, the bundled JRE version in the Chocolatey package is always, on versions <3.71, going to be 8/11.